From 9677d1b5b2871e41cbc670ae6530d2246b2d0477 Mon Sep 17 00:00:00 2001
From: Teemu Holappa <teemu.holappa@qt.io>
Date: Fri, 24 Apr 2020 11:27:20 +0300
Subject: [PATCH 036/171] Add possibility to set QNX Screen pipeline value

The pipeline value is set by window property _q_platform_qnxPipeline.

Task-number: QSR-263
Change-Id: I13536936b2335a97a6eeb5a94c4e0f552308d15e
Reviewed-by: Samuli Piippo <samuli.piippo@qt.io>
(cherry picked from commit 0c2508efb2bdeb4e96fb7726b14dea5cf2820b21)
---
 src/plugins/platforms/qnx/qqnxwindow.cpp | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/src/plugins/platforms/qnx/qqnxwindow.cpp b/src/plugins/platforms/qnx/qqnxwindow.cpp
index 1d3d609017..22a00d3d40 100644
--- a/src/plugins/platforms/qnx/qqnxwindow.cpp
+++ b/src/plugins/platforms/qnx/qqnxwindow.cpp
@@ -225,6 +225,21 @@ QQnxWindow::QQnxWindow(QWindow *window, screen_context_t context, bool needRootW
     if (windowGroup.isValid() && windowGroup.canConvert<QByteArray>())
         joinWindowGroup(windowGroup.toByteArray());
 
+    QVariant pipelineValue = window->property("_q_platform_qnxPipeline");
+    if (pipelineValue.isValid()) {
+        bool ok = false;
+        int pipeline = pipelineValue.toInt(&ok);
+        if (ok) {
+            qWindowDebug() << "Set pipeline value to" << pipeline;
+
+            Q_SCREEN_CHECKERROR(
+                screen_set_window_property_iv(m_window, SCREEN_PROPERTY_PIPELINE, &pipeline),
+                "Failed to set window pipeline");
+        } else {
+            qWindowDebug() << "Invalid pipeline value:" << pipelineValue;
+        }
+    }
+
     int debug = 0;
     if (Q_UNLIKELY(debug_fps())) {
         debug |= SCREEN_DEBUG_GRAPH_FPS;
-- 
2.31.1

